### ISSUE TITLE
[EPIC] Facility Locator iteration 2023-2024

### ISSUE THREAD

**Issue by dsinla** [department-of-veterans-affairs/va.gov-cms#11542]
The work described below is big picture and represents the larger context for multiple epics, including 

- Update Vets-API to Lighthouse v1 (Phase 1) #15344
- Facility Locator uses Lighthouse API v1 with existing functionality (Phase 2)#12907
- Update Facility Locator to allow plain language search for Lighthouse V1 implementation (Phase 3) (#14993) 
- Facility Locator Mobile Map experience #16001
- Facility Locator iteration: Provider core training/PPMS data #14225
- Update Facility Locator API to search by state #14915
---

## Background: 
In a 3/7/23 meeting with Dave Conlon we went over updates that we would like to make to facility locator. The user stories for those requested updates are below
Meeting recording here: https://drive.google.com/drive/folders/1XbflgbldfyxrO619ZMVuEb-gpEmOP7QY?usp=share_link 

Driving motivations for this epic are:

1. Improvements to search & filtering in order to display a higher volume of Services (up to 10x)
2. State level display of all facilities within the state (primarily for NCA stakeholder purposes)
3. PPMS Provider Core Training: nice to have, no longer expected as part of the MVP due to the low adoption rate of these trainings (1-2% of all providers)

## Resources
- [Past assistive technology usability research for FL](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/facilities/facility-locator/research/user-research/screenreader-usability-study/research-findings.md#research-findings-facility-locator---assistive-technology-discovery-and-usability-study)

## Overview
The facility locator update is overall focused on improving the filtering capabilities and search results. The migration to lighthouse v1 will allow for more data to be surfaced to Veterans and we want to ensure this data can be viewed in an accessible and intuitive manner.

## Problem Statement
With the migration to lighthouse v1, there will be new data that is not optimized for display in the current implementation of the facility locator. Additionally, many of the filtering capabilities 

## Updated Filtering
1.  As a Veteran, I want to have more filtering capabilities, in part made possible by the facilities v1 migration 
2. As a Veteran, I want to have a filtering experience for Service type that is usable and consistent whether I am selecting from 5 items or 800

**Implementation Notes:**
The v1 migration and current experience overall creates multiple cases we need to plan for:
- VA Health facility type will grow from from 19 to over 90 service types 
- Vet Center facility type will grow from from 0 to about 20 service types 
- VA benefits facility type will have an unknown number of service types 
- National Cemeteries will not ever list a service type, the current experience simply shows this dropdown as grey-ed out, is this the experience we want 
- Selecting the `community providers in VA's network` facility type results in over 800 (unusable) type-ahead options, how can this experience be made better
- Consider the current filter label names for the filters, they may need to be updated 
- The ability to filter by Modality (future state, in-person, virtual, at home) needs to be implemented
- Smart filtering by associated terms will be limited to the content of any given Service taxonomy term. [More details](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11542#issuecomment-1613718211)

Other ??? (unknown, but a pattern that will allow us to scale)
Urgent care, emergency, and pharmacy that show the "mashups" of 3 (all, VA-only, Comm-care only)
Benefits with TBD number

### **User stories**
- [ ] As a Veteran, I want to have more filtering capabilities, in part made possible after the facilities v1 migration 
- [ ] As a Veteran, I want to have a filtering experience for Service type that is usable and consistent whether I am selecting from 5 items or 800
- [ ] As a Facilities Product Manager, I want to be able to add several new filters to facility locator without needing to go through another design review (there should be a new design pattern that considers scalability of number of filters)
- [ ] As a Veteran, I want the filter titles to make sense to me so that I can more easily navigate the facility locator tool (background: the service type filter doesn't make sense for every use case of facility type selected, work with the content IA accessibility team to find better terms for some of the groupings, see images attached below for reference)

---

## **VA Facility by State**
1. As a Veteran, I want to be able to know what VA facilities are in a given state
2. As a Veteran, I want to be able to know what VA facilities of a certain type are in a given state (e.g. Medical, Vet Center, Benefits, Cemetery)

### **User stories**
- [ ] As a Veteran, I want to be able to know what VA facilities are in a given state
- [ ] As a Veteran, I want to be able to know what VA facilities of a certain type are in a given state (e.g. Medical, Vet Center, Benefits, Cemetery)

---

## Provider Core Training
**Design handled in this epic. implementation moved to #14225 .**
All of the following user stories in this group are related to using facility locator to search for `community providers in VA's network` the other facility types are not in the MVP scope for this effort. 
1. As a Veteran, I want to see what Providers have completed Provider Core Training
2. As a Veteran, I want to learn about Provider Core Training and what it means 
3. As a Veteran, I want to filter my search in the Facility Locator to return only providers who have received the Provider Core Training

**Implementation Notes:**
- the area for Veterans to learn about Provider core training will not be housed in facility locator, you should design a link/call to action etc. for Veterans to click or otherwise engage that links them out to this information that is housed elsewhere

For background, here is a blurb on Provider Core Training: 
Department of Veterans Affairs (VA) has created the Provider Core Training so that Veterans can be offered care from the highest quality providers available. Referring Veterans to Providers who have completed the Provider Core Training ensures that Veterans are receiving high-quality care in the community.

### **User Stories**
- [ ] As a Veteran, I want to see what Providers have completed Provider Core Training
- [ ] As a Veteran, I want to learn about Provider Core Training and what it means 
- [ ] As a Veteran, I want to filter my search in the Facility Locator to return only providers who have received the Provider Core Training

# Q3 2023
- [x] Get sign off on Designs internally (week of 6/19)
- [x] Kick off platform collab cycle with design intent meeting (week of 6/26)
- [x] Incorporate feedback from platform

Paused on usability work until post VBA Editor Research

**Comment by dsinla** [comment]
We've often spoken of a Service Locator vs a Facility Locator - to what extent do we incorporate a refocusing in this ticket?

**Comment by jilladams** [comment]
Documenting from sync with @davidconlon @mmiddaugh @kmariepat-cityfriends: 

**Decision:** Dave & Michelle do not want to consider phased roll out, as the tech debt / effort are not warranted for this iteration. We will hard cutover when it's time.

Other topics:
1. v0 vs. v1: we **do** want to feature flag behavior between V0 and V1. As protection in the event that V1 API goes left.
2. Feature flag for the front-end design changes (both designs living in parallel): Not clear if we need this. Usability testing may inform. Open questions: 
    * Can the front end still function with v0 as the data source?
    * Is the only change the number of services increasing?

We'll follow up on those elsewhere.

**Comment by kmariepat-cityfriends** [comment]
Notes about development involvement ahead of completing usability:
From a prior retro:

- FE higher level planning:
- 
- breaking down designs into components
- Architectural decisions
- how to manage versioning / cutover from v0 to v1 (as well as design updates)
- order of operations
- how to support testing from - ?
- Launch plan: % cutover / success metrics? Who decides?
- Hard to usability test bc the nature of the data, etc.
- when we get to FE task thinking, working sessions likely useful to batch thinking through the work.

**Comment by jilladams** [comment]
Some implementation nuances came up in 6/29 design review, documenting here for ref when we get to build ( with Michelle, Kamari, Swirt, Ryan, Jordan, Alexis, LauraF, Christian, and RyanT / Amanda Klausmeier): 

**Regarding how string search will work:** 
* Exact strings will be returned, e.g. search for "vision" will return all terms that include "vision"
* Algorithmic relationships between search terms will be limited to the content of the service taxonomy fields. For example: 
    * For opthalmology, fields on the VA Service taxonomy term for that include the word "vision". 
    * The data for these terms will be pulled into a JSON that will be the source material for establishing relationships. 
    * SO: when a user searches "vision", they will get "Opthalmology" as a suggested result, due to the CMS term/content relationship. 
    * If the taxonomy term does not include the word "Eye", then searching for "eye" will _not_ return Opthalmology as a result.

Meaning: we are using the content of the taxonomy to drive search results here. We are _not_ expanding scope to build a larger web of term relationships, to supplement search results.

Additional design notes are here: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/14011#issuecomment-1613716121
